test/e2e/validation_test.go

```diff
@@ -43,6 +43,15 @@ func (tc *testContext) getInstanceIDsOfNodes() ([]string, error) {
 	return instanceIDs, nil
 }
 
+// testKubeletVersion tests if the kubelet version is present
+func testKubeletVersion(t *testing.T) {
```

Comment:
Please make a single test called `testNodeMetadata` or something along those line with `testKubeletVersion` and other tests like `testWorkerLabel` where we are looping through nodeas sub tests. i.e. do table driven tests.